content/blog/2022-07-12-parity-explained/index.md

```diff
@@ -0,0 +1,248 @@
+
+{{< img src="blog-snapshot-diff.png" >}}
+
+As you can see, there are some additional parameters in the AWS snapshot, that are missing in the LocalStack response, and vice versa. 
+
+This information is very important and helps us to access, improve, and fix expected responses. 
+
+While this is a just a simple example, it showcases the power of the snapshot recording and testing. 
+
+#### Parity Tests in Action
+
+Snapshot tests should be the preferred way of writing new tests. However, some tests require additional preparation and setup to work against AWS, which naturally slows down the process of migration. 
+
+We already have some snapshot tests included in our repository. 
+In order to make the migration easier, we also added a marker `pytest.mark.skip_snapshot_verify` to temporarly disable the verification, which can be helpful when preparing or re-writing a test case. 
+
+With this marker it is also possible to specify a list of json-paths, which would disable the verification of the snapshot only for the specified paths:
+
+```python
+@pytest.mark.skip_snapshot_verify(paths=["$..User.Tags"])
+def test_iam_username_defaultname(deploy_cfn_template, iam_client, snapshot):
+    snapshot.add_transformer(snapshot.transform.iam_api())
+    snapshot.add_transformer(snapshot.transform.cloudformation_api())
+
+    template = json.dumps(
+        {
+            "Resources": {
+                "DefaultNameUser": {
+                    "Type": "AWS::IAM::User",
+                }
+            },
+            "Outputs": {"DefaultNameUserOutput": {"Value": {"Ref": "DefaultNameUser"}}},
+        }
+    )
+    stack = deploy_cfn_template(template=template)
+    user_name = stack.outputs["DefaultNameUserOutput"]
+    assert user_name
+
+    get_iam_user = iam_client.get_user(UserName=user_name)
+    snapshot.match("get_iam_user", get_iam_user)
+```
+
+This can be helpful in a situation like in the test case `test_iam_username_defaultname` above: one attribute is added in the output, e.g., it is not returned by AWS, but returned by LocalStack. 
+
+This strategy allows us to collect and outline deviations, while making sure that existing behavior is not broken accidentially. 
+
+
+## Outlook
+With the new ASF framework we also introduced a metric collection utility. It enables us to collect details during the test execution, including used parameter values, exceptions etc.
+This information will help us to increase test coverage and consequently improve the parity with AWS. 
+
+Additionally, we will provide regular metric updates and insights about supported services and operations. Thus our communication will be more transparent in terms of implemented APIs, and further improve the confidence overall. 
+
+We hope you are as excited as we are about the new snapshot testing framework - and if you are contributor: we appreciate your support with enhancing the AWS parity!
```

Comment:
it would be great if we could make the concluding sentence encompass the entire blog post, not just the snapshot testing framework (even though i *am* really excited about it :D)

Reply:
The last sentence is the most difficult one 🙂 What about: 
"We hope you are as excited as we are about our AWS Framework, that ensures latest compatibility with AWS, and the new snapshot testing framework, that will help us to write validated test cases."